web/src/main/java/com/microsoft/applicationinsights/web/extensibility/initializers/WebAppNameContextInitializer.java

```diff
@@ -0,0 +1,23 @@
+package com.microsoft.applicationinsights.web.extensibility.initializers;
+
+import com.microsoft.applicationinsights.common.CommonUtils;
+import com.microsoft.applicationinsights.extensibility.ContextInitializer;
+import com.microsoft.applicationinsights.extensibility.context.CloudContext;
+import com.microsoft.applicationinsights.telemetry.TelemetryContext;
+
+public class WebAppNameContextInitializer implements ContextInitializer {
+
+    private String appName;
+
+    public void setAppName(String appName) {
+        this.appName = appName;
+    }
+
+    @Override
+    public void initialize(TelemetryContext context) {
+        if (!CommonUtils.isNullOrEmpty(appName)) {
+            CloudContext cloud = context.getCloud();
+            cloud.setRole(appName);

```

Comment:
This part is specifically related to what I commented about AppMap component name. When you set this from the Filter this would override the appName which springboot gives. 
Specifically when we think about converting TelemetryInitializer to ContextInitializer (the one in Starter)